src/commerce-handler.js

```diff
@@ -61,6 +116,182 @@ CommerceHandler.prototype.logCommerceEvent = function(event) {
             21: ProductRemoveFromWishlist,
             22: ProductImpression
         */
+    let events = [];
+
+    switch (event.EventCategory) {
+        case ProductActionTypes.Impression:
+            events = buildImpressionEvents(event);
+            break;
+        case PromotionType.PromotionClick:
+        case PromotionType.PromotionView:
+            events = buildPromotionEvents(event);
+            break;
+        default:
+            events = buildProductActionEvents(event);
+    };
+
+    events.forEach((event) => {
+        window.heap.track(event.Name, event.Properties);
+    });
+};
+
+function buildImpressionEvents(event) {
+    let events = [];
+
+    if (event.ProductImpressions.length > 0) {
+        event.ProductImpressions.forEach((impression) => {
```

Comment:
As much as I personally love fat arrow functions (`=>`) you should go old school here and use `function () {` instead. Our kits don't have transpiling set up so this may not work on older browsers.